pkg/writer/serializer_cdx14.go

```diff
@@ -138,25 +138,43 @@ func (s *SerializerCDX14) Serialize(opts options.Options, bom *sbom.Document) (i
 
 		// Now add al nodes we have not yet positioned
 		for _, c := range components {
-			if _, ok := addedDict[c.Ref]; ok {
+			if _, ok := addedDict[c.BOMRef]; ok {
 				continue
 			}
-			doc.Components = append(doc.Components, *c)
+			*doc.Components = append(*doc.Components, *c)
 		}
 
 		// Add components without refs
 		for _, c := range refless {
-			doc.Components = append(doc.Components, *c)
+			*doc.Components = append(*doc.Components, *c)
 		}
 	}
 	return doc, nil
 }
 
 func (s *SerializerCDX14) Render(opts options.Options, doc interface{}, wr io.Writer) error {
 	logrus.Debug("Writing SBOM in CycloneDX to STDOUT")
-	encoder := json.NewEncoder(wr)
-	encoder.SetIndent("", strings.Repeat(" ", opts.Indent))
-	if err := encoder.Encode(doc.(*cdx14.Document)); err != nil {
+	encoder := cdx14.NewBOMEncoder(wr, cdx14.BOMFileFormatJSON)
+	if doc == nil {
+		return fmt.Errorf("no doc found")
+	}
+
+	if opts.Format.Type() != options.CDXFROMAT {
+		return fmt.Errorf("unsupported options, %v", opts)
+	}
+
+	encoder.SetPretty(true)
+
+	cdxVersion := cdx14.SpecVersion1_4
+	// 2DO - not sure if we should implicit switch case the suppported version or simply transform
```

Comment:
Can you explain this a little bit more?

Reply:
Yeah I think the cdx 1.4 serializer should always return 1.4,  we can do a super simple writer for older versions that uses the same serializer and just have it return an older format.